backend/dal/Repositories/ActivityRepository.cs

```diff
@@ -87,22 +90,26 @@ public PimsActivityInstance Add(PimsActivityInstance instance)
         public PimsActivityInstance Update(PimsActivityInstance instance)
         {
             instance.ThrowIfNull(nameof(instance));
-            GetById(instance.ActivityInstanceId);
-            if (instance.ActivityTemplate != null)
-            {
-                Context.Entry(instance.ActivityTemplate).State = EntityState.Unchanged;
-            }
-            foreach (PimsActivityInstanceDocument activityInstanceDocument in instance.PimsActivityInstanceDocuments)
-            {
-                Context.Entry(activityInstanceDocument).State = EntityState.Unchanged;
-            }
+            var currentActivity = this.Context.PimsActivityInstances
+                .FirstOrDefault(x => x.ActivityInstanceId == instance.Id) ?? throw new KeyNotFoundException();
+            this.Context.Entry(currentActivity).CurrentValues.SetValues(instance);
```

Comment:
it is more readable, but I guess it does not update the related properties if we would want that.

Reply:
yes, that would need to be done explicitly (perhaps more akin to what I'm doing with property updates here).